boot/zephyr/decompression.c

```diff
@@ -282,8 +288,41 @@ int bootutil_img_hash_decompress(struct enc_key_data *enc_state, int image_index
                 offset_zero_check = 0;
             }
 
+            /* Copy data to secondary buffer for calculating hash */
             if (output_size > 0) {
-                bootutil_sha_update(&sha_ctx, output, output_size);
+                if (hdr->ih_flags & IMAGE_F_COMPRESSED_ARM_THUMB_FLT) {
+                    /* Run this through the ARM thumb filter */
+                    uint32_t offset_arm_thumb = 0;
+                    uint8_t *output_arm_thumb = NULL;
+                    uint32_t output_size_arm_thumb = 0;
+
+                    rc = compression_arm_thumb->decompress(NULL, output, output_size, last_packet,
+					                   &offset_arm_thumb, &output_arm_thumb,
+                                                           &output_size_arm_thumb);
+
+                    if (rc) {
+                        BOOT_LOG_ERR("Decompression error: %d", rc);
+                        rc = BOOT_EBADSTATUS;
+
```

Comment:
The empty line looks odd; somehow I think that it should be above `rc = ` as the `rc = ` and `goto finish` serve as "return".

Reply:
Removed the lines